### ISSUE TITLE
Multi monitor app straddling

### ISSUE THREAD

**Issue by droc12345** [WayfireWM/wayfire#1335]
Not sure if this is a bug, or not, so I marked as other.
I'm running git latest version, (nov 6 sync) 

I was just playing with my multi-monitor setup (1 4k, 1 1080p) and noticed behavior different than the way it used to be.
The way it worked in the past was if I grab a window, and pulled it part way across the monitors, so that part of the window is one monitor and the other monitor shows the rest of the window. 

When I'm holding down the mouse and moving the window across the monitors, you see a faint window that shows on both monitors, but as soon as I release the mouse button, then only that portion of the window is shown on whichever monitor the mouse focus is on. 

Expected behavior is that I should see both parts of the window, a part on each monitor.

**Comment by ammen99** [comment]
> The way it worked in the past was if I grab a window, and pulled it part way across the monitors, so that part of the window is one monitor and the other monitor shows the rest of the window.

This has never been the case.

> Expected behavior is that I should see both parts of the window, a part on each monitor.

This isn't the official expected behavior. Wayfire's outputs are meant to be independent, and each view can be on a single monitor. There are upcoming changes which will make it possible to sanely write a plugin which provides what you want though, and I'll probably try to write it myself to validate the design in core.

TLDR; Wayfire has never worked this way, but it will be possible in the future (but I can give no ETA)

**Comment by droc12345** [comment]
I'm pretty sure I've seen it work that way in the past, but that was a while back, maybe wlroots 13.0 days.

Another problem.
Now the fonts are very blurry on the monitor with scaling applied, that was not the case in the past either. 
In other words, if I start a terminal on the 4k monitor and the 1080p monitor they should appear similar (given the obvious difference between PPI/DPI). Now the upscaled 1080p is very blurry. 

I use primarily two compositors, wayfire and labwc.  Labwc still shows an app split across 2 monitors and the fonts are not blurry on the scaled monitor.

**Comment by ammen99** [comment]
> Another problem. Now the fonts are very blurry on the monitor with scaling applied, that was not the case in the past either. In other words, if I start a terminal on the 4k monitor and the 1080p monitor they should appear similar (given the obvious difference between PPI/DPI). Now the upscaled 1080p is very blurry.

Please open separate issues for .. separate issues :)

> I'm pretty sure I've seen it work that way in the past, but that was a while back, maybe wlroots 13.0 days.

And I'm 100% sure this has never worked this way in Wayfire. The code has never been written to support something like this. My guess is that you're confusing Wayfire and Labwc.

**Comment by knecht** [comment]
> There are upcoming changes which will make it possible to sanely write a plugin which provides what you want though, and I'll probably try to write it myself to validate the design in core.

Is there some other issue to track this effort? I'm asking because this one is closed and the sentence above indicates there is still hope. This issue was the reason why I switched back from wayfire to x11/compiz again a couple of months ago.
I have 3 monitors attached and I would love to use wayfire again. But not being able to use a window crossing the monitor border is much worse for me than the many small bugs of compiz. 

If there is something other people (like me) could do to help, some hints what to do and where to start might be useful.

**Comment by knecht** [comment]
see also #1497
